ta/pkcs11/src/processing_ec.c

```diff
@@ -646,3 +646,100 @@ size_t ecdsa_get_input_max_byte_size(TEE_OperationHandle op)
 		return 0;
 	}
 }
+
+enum pkcs11_rc pkcs2tee_param_ecdh(struct pkcs11_attribute_head *proc_params,
+				   void **pub_data, size_t *pub_size)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct serialargs args = { };
+	uint32_t temp = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	/* Skip KDF */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	/* Shared data size, shall be 0 */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || temp)
+		return rc;
+
+	/* Public data size and content */
+	rc = serialargs_get(&args, &temp, sizeof(uint32_t));
+	if (rc || !temp)
+		return rc;
+
+	*pub_size = temp;
+
+	rc = serialargs_get(&args, &temp, sizeof(uint8_t));
+	if (rc)
+		return rc;
+
+	if (temp != 0x02 && temp != 0x03 && temp != 0x04)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	if (temp != 04) {
+		EMSG("DER compressed public key format not yet supported");
+		return PKCS11_CKR_ARGUMENTS_BAD;
+	}
+
+	*pub_size -= sizeof(uint8_t);
+
+	if (*pub_size >= 0x80) {
+		EMSG("DER long definitive form not yet supported");
+		return PKCS11_CKR_ARGUMENTS_BAD;
+	}
+
+	rc = serialargs_get_ptr(&args, pub_data, *pub_size);
+	if (rc)
+		return rc;
+
+	if (serialargs_remaining_bytes(&args))
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	return PKCS11_CKR_OK;
+}
+
+enum pkcs11_rc pkcs2tee_algo_ecdh(uint32_t *tee_id,
+				  struct pkcs11_attribute_head *proc_params,
+				  struct pkcs11_object *obj)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct serialargs args = { };
+	uint32_t kdf = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	rc = serialargs_get(&args, &kdf, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	/* Remaining arguments are extracted by pkcs2tee_param_ecdh */
+	if (kdf != PKCS11_CKD_NULL)
```

Comment:
A last comment: maybe  worth to add a debug trace stating this implementation only supports CKD_NULL derivation, known the PKCS#11 specification also defines CKD_SHA1_KDF scheme.
E.g.:
```c
	if (kdf != PKCS11_CKD_NULL) {
		DMSG("Only support CKD_NULL key derivation for ECDH");
		return PKCS11_CKR_MECHANISM_PARAM_INVALID;
	}

Reply:
Sure I'll add that comment. The thing is that I didn't have to do much on this commit - it only took me a few hours to re-assemble the pieces you and Vesa did and then just validated them. 
I'll wait for @vesajaaskelainen  comment on the tag (I dont think we use Co-developed-by in OP-TEE? should I do that instead?)

Reply:
> (I dont think we use Co-developed-by in OP-TEE? should I do that instead?)

`Co-developed-by:` has been used occasionally, it's a good way to acknowledge other people's work in my opinion.
```
$ git log --oneline --grep=Co-developed-by | wc -l
35
```

Reply:
ah yeah, ok, let me do that instead then.

Reply:
As for me, I let you choose between either
```
Co-developed-by: Etienne Carriere <etienne.carriere@linaro.org>
Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
```
or
```
Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>
```
at your will. But note that both are somewhat exclusive.

Reply:
ok, so I'll add you as reviewer and @vesajaaskelainen as Co-developed-by (unless he prefers the review tag as well).